codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionTest.java

```diff
@@ -425,7 +425,7 @@ public void reserveWithPushDisallowedShouldThrow() throws Http2Exception {
     @Test(expected = Http2Exception.class)
     public void goAwayReceivedShouldDisallowCreation() throws Http2Exception {
         server.goAwayReceived(0, 1L, Unpooled.EMPTY_BUFFER);
-        server.remote().createStream(3, true);
+        server.local().createStream(3, true);
```

Comment:
why this change ?

Reply:
the unit test was incorrect and highlighted a bug addressed by this PR.

The test simulates receiving a go away frame, then created a stream from the remote endpoint, and expected an error. However this is not correct. Receiving a goaway frame should prevent the *local* endpoint from creating streams and have no effect on the *remote* endpoint.

Reply:
I added some additional tests to catch all the cases and clarify.